templates/puppet/property/resourceref.rb.erb

```diff
@@ -98,21 +99,16 @@ module Google
           unless /<%= property.resource_ref.regex_url.gsub('/', '\/') -%>/.match(@title)
             # We'll asssmble the self_link for the user.
             # We need to get the project name to assemble the self_link
-            projects = Google::ObjectStore.instance
-                                          .resources
-                                          .reject { |name, _a| name == :gauth_credential }
-                                          .map { |_name, array| array.map(&:exports) }
-                                          .flatten
-                                          .map { |x| x[:project] }
-                                          .compact
-                                          .uniq
-            return "<%= property.resource_ref.self_link_url.join.gsub('{{name}}', '#{@title}').gsub('{{project}}', '#{projects[0]}') -%>" \
-              if projects.length == 1
-            raise ["Your Puppet manifest contains multiple projects.",
-                   "We cannot determine which project you wish to use.",
-                   "Please replace #{@title} with a full URL of the form:",
-                   "<%= property.resource_ref.self_link_url.join -%>"
-                  ].join(' ')
+<%
+  # We need to grab values from the @resource.
+  # This substitution will grab the values from @resource and convert to String (in case it's an object)
+  url = property.resource_ref.self_link_url.join
+                .gsub('{{name}}', '#{@title}')
+                .gsub('}}', '].value.to_s}')
+                .gsub('{{', '#{@resource.parameters[:')
```

Comment:
I find this a little spooky.  I can't prove that it's right in my head, but it looks about right - the idea here is that
`{{foo}}` becomes `#{@resource.parameters[:foo].value.to_s}`?  Is the `.to_s` necessary, since it's being string-substituted with the `#{}`?

Reply:
Yes, I should add a comment explaining what I'm doing with that. You have the idea exactly right though. Any ideas how to clean up that code?

You're probably right that the .to_s is unnecessary. I'll do a quick test and update.